### DISCUSSION TITLE
Error 403: access_denied

### DISCUSSION THREAD

**Discussion by AseKarlsson** [YorkshireIoT/ha-google-fit#121] [General]
Hi,
Was interested to add this integration into my Home Assistant.
Have not configured any Google API before, so maybe I have missed something because I end up with this error message:

![image](https://github.com/YorkshireIoT/ha-google-fit/assets/80900477/ac714fee-e548-4753-a439-58648e765cab)

I do not have Home Assistant Cloud, only local Home Assistant installation, if that is the issue?

Extra info:

![image](https://github.com/YorkshireIoT/ha-google-fit/assets/80900477/0114d5e7-d67a-4ddf-8d77-5d3f9270b27c)

Thanks,
Ase

**Comment by YorkshireIoT** [comment]
Hi @AseKarlsson,

I haven't seen this before. Just to clairfy something quickly though, it won't be because you don't have Home Assistant Cloud.

Can I first ask you double check your setup process using these screenshots from the [README](https://github.com/YorkshireIoT/ha-google-fit#setup-example-screenshots)? You may need to first delete your credentials first, instructions [here](https://github.com/YorkshireIoT/ha-google-fit#reconfiguring-badly-configured-credentials).

For this config flow to work, it's crucial that the link [configured here](https://github.com/YorkshireIoT/ha-google-fit/blob/main/res/link_account.png) points exactly to your instance.

**Comment by AseKarlsson** [comment]
Hi @YorkshireIoT,
Tried to delete the credentials, and tried again.
Instead of this:
![image](https://github.com/YorkshireIoT/ha-google-fit/assets/80900477/a85584dc-a3be-44d7-a8db-d068629d0b8d)

 I get this:
![image](https://github.com/YorkshireIoT/ha-google-fit/assets/80900477/4f74adf9-95a3-4c0c-b7c7-6ee04ed206a2)

and never come to this:
![image](https://github.com/YorkshireIoT/ha-google-fit/assets/80900477/63dbcd74-64bd-47bf-8409-d7e90e1eca55)

**Comment by YorkshireIoT** [comment]
Can you try this suggestion here? It seems to be exactly the problem you're having:
https://community.home-assistant.io/t/nest-integration-access-blocked/483375/3

**Reply by AseKarlsson** [reply]
That was it, thanks a lot!
